host_core/test/host_core/benchmark/echo_test.exs

```diff
@@ -0,0 +1,219 @@
+defmodule HostCore.BenchmarkTest do
+  # We'd rather not run this test asynchronously because it's a benchmark. We'll get better
+  # results if this is the only test running at the time.
+  use ExUnit.Case, async: false
+
+  alias HostCore.Actors.ActorRpcServer
+  alias HostCore.Actors.ActorSupervisor
+  alias HostCore.WasmCloud.Native
+  alias HostCore.Linkdefs.Manager
+  alias HostCore.Providers.ProviderSupervisor
+
+  import HostCoreTest.Common, only: [cleanup: 2, standard_setup: 1, request_http: 2]
+
+  @echo_key HostCoreTest.Constants.echo_key()
+  @echo_path HostCoreTest.Constants.echo_path()
+  @echo_wasi_key HostCoreTest.Constants.echo_wasi_key()
+  @echo_wasi_path HostCoreTest.Constants.echo_wasi_path()
+
+  @httpserver_path HostCoreTest.Constants.httpserver_path()
+  @httpserver_key HostCoreTest.Constants.httpserver_key()
+  @httpserver_contract HostCoreTest.Constants.httpserver_contract()
+  @httpserver_link HostCoreTest.Constants.default_link()
+
+  describe "Benchmarking actor invocations" do
+    setup :standard_setup
+
+    test "load test with echo wasm32-unknown actor", %{
+      :evt_watcher => evt_watcher,
+      :hconfig => config,
+      :host_pid => pid
+    } do
+      on_exit(fn -> cleanup(pid, config) end)
+
+      # TODO: make configurable w/ benchee
+      num_actors = 1
+      parallel = [1, 10]
+
+      {msg, inv, port} = setup_echo_test(config, evt_watcher, @echo_key, @echo_path, num_actors)
+
+      # very noisy debug logs during bench
+      {:ok, _okay} = HTTPoison.start()
+      Logger.configure(level: :info)
+
+      test_config = %{
+        "direct_echo_request" => fn ->
+          ActorRpcServer.request(msg)
+        end,
+        "nats_echo_request" => fn ->
+          config.lattice_prefix
+          |> HostCore.Nats.rpc_connection()
+          |> HostCore.Nats.safe_req(msg.topic, inv, receive_timeout: 2_000)
+        end,
+        "http_echo_request" => fn ->
+          {:ok, _resp} = request_http("http://localhost:#{port}/foo/bar", 1)
+        end
+      }
+
+      # Run the test at a few specified levels of parallelism, allowing for some warmup time to let compute calm down
+      parallel
+      |> Enum.each(fn parallel ->
+        IO.puts("Benchmarking with #{num_actors} actors and #{parallel} parallel requests")
+
+        Benchee.run(test_config,
+          warmup: 1,
+          time: 5,
+          parallel: parallel
+          # save: [path: "./echo_wasm32_unknown_#{num_actors}_#{parallel}.txt"] # TODO: this should be how we compare
+        )
+      end)
```

Comment:
Could some of this scaffolding be extracted and shared between the 3 tests? I'm thinking:
- setting the logger level at the beginning and end
- take a `test_config`, `num_actors`, and `parallel` and run the benchee stuff

Reply:
Yeah let me see if I can define some of this stuff in helpers